src/library_memfs.js

```diff
@@ -341,7 +341,7 @@ mergeInto(LibraryManager.library, {
         var contents = stream.node.contents;
         // Only make a new copy when MAP_PRIVATE is specified.
         if ( !(flags & {{{ cDefine('MAP_PRIVATE') }}}) &&
-              (contents.buffer === buffer || contents.buffer === buffer.buffer) ) {
+              (contents.buffer === buffer) ) {
```

Comment:
Yeah, that would work as well I think. Although there is a risk that user nukes the `Module` object some way, so since this is already referring to `HEAPU8` below, keeping to refer to that would be consistent(?) No big preference either way though.